### PR TITLE
rebuild v 0.2.7:

### PR DESCRIPTION
**Pull request by akabanovs**
- build number increased.
 - patch added to remove entrypoints dependency.
- python pinning modified

The issue was actually observed when doing some tests for streamlit-faker and not streamlit-extras. But it is streamlit-extras that needs some patching. 

streamlit-extras and streamlit are both used by streamlit-faker. With the latest streamlit 1.24.1  (https://github.com/AnacondaRecipes/streamlit-feedstock/pull/5), streamlit-faker started to fail:

`Traceback (most recent call last):
  File "/Users/antonkabanovs/wkspaces/streamlit-faker/streamlit-faker_1689070809379/test_tmp/run_test.py", line 2, in <module>
    import streamlit_faker
  File "/Users/antonkabanovs/wkspaces/streamlit-faker/streamlit-faker_1689070809379/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.10/site-packages/streamlit_faker/__init__.py", line 3, in <module>
import: 'streamlit_faker'
    from .chart import StreamlitChartProvider
  File "/Users/antonkabanovs/wkspaces/streamlit-faker/streamlit-faker_1689070809379/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.10/site-packages/streamlit_faker/chart.py", line 7, in <module>
    from streamlit_extras import altex
  File "/Users/antonkabanovs/wkspaces/streamlit-faker/streamlit-faker_1689070809379/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.10/site-packages/streamlit_extras/altex/__init__.py", line 5, in <module>
    import entrypoints
ModuleNotFoundError: No module named 'entrypoints'`

So the problem is that now nothing provides entrypoints. 
It turns out that the old streamlit v 1.16.0 used altair with version range >=3.2.0<5, resolving into 4.1.0 and taken from noarch:
altair-4.1.0-py_1.tar.bz2
    ├──entrypoints
    ├──jinja2
    ├──jsonschema
    ├──numpy >=0.18
    ├──pandas
    ├──python >=3.6
    └──toolz
    
Now, the new streamlit 1.24.1, which is going to be delivered,  uses altair with range >=4.0,<6. This range resolves into altair v 5.0.1 available on osx-arm64 arch directly. Here are it's runtime dependencies:
altair-5.0.1-py311hca03da5_0.tar.bz2
    ├──jinja2
    ├──jsonschema >=3.0
    ├──numpy
    ├──pandas >=0.18
    ├──python >=3.11,<3.12.0a0
    └──toolz
  
It can be seen that altair 5.0.1 does not have entrypoints as a dependency (and it should not, I checked) while altair 4.1.0 had entrypoints as a runtime dependency.
 
It is a bit of a mess. In short - altair 4.1.0  provides entrypoints (which is why previously there was no issue), but not the altair v 5.0.1 that the new streamlit 1.24.1 requires, which as a result breaks streamlit-extras and streamlit-faker.

It was decided to patch up streamlit-extras to remove entrypoints dependency, as this package is in maintenance mode and just used in a few places. Also, entrypoints is not explicitly mentioned as a dependency in streamlit-extras.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,22 +8,28 @@ package:
...
+    - m2-patch  # [win]
   host:
     - python
     - poetry-core
     - pip
     - setuptools
     - wheel
   run:
-    - python <3.9.7|>3.9.7,<4.0
+    - python <3.9.7,>3.9.7,<4.0
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:34]: python != 3.9.7, would that work ?

File: recipe/patches/0001-remove-the-usage-of-entrypoints.patch
Comment (M-Waszkiewicz-Anaconda) [inline recipe/patches/0001-remove-the-usage-of-entrypoints.patch:1]: why we need to remove entrypoint dependency ?,  do we miss such package in defaults ?

Comment (skupr-anaconda) [reply recipe/patches/0001-remove-the-usage-of-entrypoints.patch:1]: I agree that some explanation/comment is needed

File: recipe/meta.yaml (hunk 1 above)
Comment (akabanovs) [reply recipe/meta.yaml:34]: Will probably work. Just wanted to keep the amount of changes to a minimum. | might have been a typo.

File: recipe/patches/0001-remove-the-usage-of-entrypoints.patch
Comment (akabanovs) [reply recipe/patches/0001-remove-the-usage-of-entrypoints.patch:1]: Please see the description:
https://github.com/AnacondaRecipes/streamlit-extras-feedstock/pull/3#issue-1800750337

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Pull request merged**